bentoml/yatai/grpc_stream_utils.py

```diff
@@ -0,0 +1,138 @@
+import os
+import io
+import tarfile
+from bentoml.yatai.proto.repository_pb2 import UploadBentoRequest, DownloadBentoResponse
+
+
+DEFAULT_FILE_CHUNK_SIZE = 1024 * 8
+
+
+class FileStream:
+    def __init__(self):
+        self.buffer = io.BytesIO()
+        self.offset = 0
+
+    def write(self, s):
+        self.buffer.write(s)
+        self.offset += len(s)
+
+    def tell(self):
+        return self.offset
+
+    def close(self):
+        self.buffer.close()
+
+    def read_value(self):
+        try:
+            return self.buffer.getvalue()
+        finally:
+            self.buffer.close()
```

Comment:
is there a `flush` method for BytesIO?

Reply:
`bytesio.close` does the flushing and once closed, any operations will raise exception. It is recommend.